python/test/unit/language/test_core.py

```diff
@@ -6422,6 +6422,70 @@ def test_convert2d(M, N, src_layout, interm_layout, dst_layout, dtype, device, t
     torch.testing.assert_close(z, x, rtol=0, atol=0)
 
 
+src_layouts = [BlockedLayout([1, 1], [1, 64], [1, 1], [0, 1], [1, 1], [1, 1], [0, 1])]
+
+dst_layouts = [
+    LinearLayout([[0, 32]], [[0, 1], [0, 2], [0, 4], [0, 8], [0, 16], [1, 0]], [], []),
+    LinearLayout([[1, 0], [0, 32]], [[0, 1], [0, 2], [0, 4], [0, 8], [0, 16], [2, 0]], [], []),
+    LinearLayout([[1, 0], [2, 0], [0, 32]], [[0, 1], [0, 2], [0, 4], [0, 8], [0, 16], [4, 0]], [], [])
+]
+
+
+@pytest.mark.parametrize("M, dst_layout", [[2, dst_layouts[0]], [4, dst_layouts[1]], [8, dst_layouts[2]]])
+@pytest.mark.parametrize("src_layout", [src_layouts[0]])
+@pytest.mark.parametrize("N", [64])
+@pytest.mark.parametrize("dtype", ['float8e5', 'float16', 'float32', 'int64'])
+def test_convert_permlane_swap(M, N, src_layout, dst_layout, dtype, device, tmp_path: pathlib.Path):
+    if not is_hip_cdna4():
```

Comment:
Given this is AMD specific test, we might want to put it under `third_party/amd/python/test/`.

Reply:
Sorry, one more thing, we need to register the test here https://github.com/triton-lang/triton/blob/4b92505335a579f78c03bcf1c12457ca5b18e683/.github/workflows/integration-tests-amd.yml#L123. Probably best to make it something like `pytest --capture=tee-sys -rfs -n 8 third_party/amd/python/test/ --ignore=test_scalarize_packed_fops.py --ignore=test_address_sanitizer.py` to let it automatically pick up new tests there.